openpgp/keys.go

```diff
@@ -500,13 +520,26 @@ EachPacket:
 					// We really should warn that there was a failure here. Not raise an error
 					// since this really shouldn't be a fail-stop error.
 				}
-			} else if pkt.SigType == packet.SigTypeKeyRevocation {
-				// These revocations won't revoke UIDs as handled above, so lookout!
-				revocations = append(revocations, pkt)
+			} else if current != nil && pkt.SigType == packet.SigTypeIdentityRevocation {
+				if err = e.PrimaryKey.VerifyUserIdSignature(current.Name, e.PrimaryKey, pkt); err == nil {
+					current.Revocation = pkt
+
+					// Remove current from Identities, because the revocation signature
+					// might come after the certification signature.
+					delete(e.Identities, current.Name)
```

Comment:
Yes. Otherwise whether revoked uid lands in Identities or BadIdentities is decided by order of signatures. If revocation comes first, it'll go to BadIdentities, if certification comes first, it'll go to Identities (but both cases will set `current.Revocation` field).